src/napari_phasors/lifetime_tab.py

```diff
@@ -0,0 +1,552 @@
+
+        # Clamp to valid range
+        min_val = max(0.0, min(min_val, self.max_lifetime or 0))
+        max_val = max(0.0, min(max_val, self.max_lifetime or 0))
+
+        # Ensure min < max with small epsilon
+        if min_val >= max_val:
+            max_val = min_val + 0.01
+
+        min_slider = int(min_val * self.lifetime_range_factor)
+        max_slider = int(max_val * self.lifetime_range_factor)
+        self.lifetime_range_slider.setValue((min_slider, max_slider))
+        self._on_lifetime_range_changed((min_slider, max_slider))
+
+    def _on_lifetime_max_edit(self):
+        min_val = float(self.lifetime_min_edit.text())
+        max_val = float(self.lifetime_max_edit.text())
+
+        # Clamp to valid range
+        min_val = max(0.0, min(min_val, self.max_lifetime or 0))
+        max_val = max(0.0, min(max_val, self.max_lifetime or 0))
+
+        # Ensure min < max with small epsilon
+        if max_val <= min_val:
+            min_val = max_val - 0.01 if max_val > 0.01 else 0.0
+
+        min_slider = int(min_val * self.lifetime_range_factor)
+        max_slider = int(max_val * self.lifetime_range_factor)
+        self.lifetime_range_slider.setValue((min_slider, max_slider))
+        self._on_lifetime_range_changed((min_slider, max_slider))
+
+    def plot_lifetime_histogram(self):
+        """Plot the histogram of the lifetime data as a line plot."""
+        if self.lifetime_data is None:
+            self.histogram_widget.hide()
+            return
+        if self.parent_widget._labels_layer_with_phasor_features is None:
+            self.histogram_widget.hide()
+            return
+        if self.parent_widget.harmonic is None:
+            self.histogram_widget.hide()
+            return
+
+        flattened_data = self.lifetime_data.flatten()
+        flattened_data = flattened_data[~np.isnan(flattened_data)]
+        flattened_data = flattened_data[flattened_data > 0]
+        flattened_data = flattened_data[np.isfinite(flattened_data)]
+
+        if len(flattened_data) == 0:
+            self.hist_ax.clear()
+            self.hist_ax.text(
+                0.5,
+                0.5,
+                'No valid data',
+                transform=self.hist_ax.transAxes,
+                ha='center',
+            )
+            self.hist_fig.canvas.draw_idle()
+            self.histogram_widget.show()
+            return
+
+        self.counts, self.bin_edges = np.histogram(flattened_data, bins=300)
+        self.bin_centers = (self.bin_edges[:-1] + self.bin_edges[1:]) / 2
+
+        self._update_lifetime_histogram()
+        self.histogram_widget.show()
+
+    def create_lifetime_layer(self):
+        """Create or update the lifetime layer for all harmonics."""
+        if self.lifetime_data is None:
+            return
+        lifetime_layer_name = f"{self.lifetime_type_combobox.currentText()} Lifetime: {self.parent_widget.image_layer_with_phasor_features_combobox.currentText()}"
```

Comment:
Removes duplicated "Lifetime" in layer names
```suggestion
        lifetime_layer_name = f"{self.lifetime_type_combobox.currentText()}: {self.parent_widget.image_layer_with_phasor_features_combobox.currentText()}"
```